File: python_mid_june_2024/11. Container With Most Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i, j = 0, len(height) - 1
        area = 0

        while i < j:
            h = min(height[i], height[j])
            area = max(area, (j - i)*h)

            if height[i] < height[j]:
                i += 1
            else:
                j -= 1
        
        return area

File: java_july/11. Container With Most Water.java
class Solution {
    public int maxArea(int[] height) {
        int i=0, j=height.length-1, area = 0;

        while (i < j) {
            int h = Math.min(height[i], height[j]);
            area = Math.max(area, h * (j-i));
            
            while (height[i] <= h && i < j) i++;
            while (height[j] <= h && i < j) j--;
        }

        return area;
    }
}